app/src/common/shared/com/igalia/wolvic/search/SearchEngineWrapper.java

```diff
@@ -66,13 +56,17 @@ SearchEngineWrapper get(final @NonNull Context aContext) {
     private SearchSuggestionClient mSuggestionsClient;
     private SharedPreferences mPrefs;
     private boolean mAutocompleteEnabled;
+    private SearchEngineManager mSearchEngineManager;
+    private HashMap<String, SearchEngine> mSearchEnginesMap;
 
     private SearchEngineWrapper(@NonNull Context aContext) {
-        mContext = aContext;
+        mContext = aContext.getApplicationContext();
         mPrefs = PreferenceManager.getDefaultSharedPreferences(mContext);
         mAutocompleteEnabled = SettingsStore.getInstance(mContext).isAutocompleteEnabled();
 
-        setupSearchEngine(aContext, EMPTY);
+        String preferredSearchEngineId = SettingsStore.getInstance(mContext).getSearchEngineId();
+
+        setupSearchEngine(aContext, preferredSearchEngineId);
```

Comment:
So the current code does not select any preferred engine, why do we need to specify one now?

Reply:
The current code selects the preferred search engine based on the language and location of the user, without the user being able to make a choice.

My intention with this PR was:

- use the language and location to get the list of available search engines
- if the user has _not_ given a preference, we will take the default option provided by the search component
- if the user _has_ given a preference, we will use that.

(I will take another look at the code to verify that this is actually the case).

Reply:
I had a look in detail and the implementation in this PR does not always work in the way I described above:

1) The first time that you launch Wolvic, the stored preference is empty so we will pick the default search engine for your locale/location and store its `id`.

2) When you select a different search engine, we will store its `id` (replacing the previous value).

3) When locale or location change, the list of available search engines may change as well. If the stored `id` (which came from points 1 or 2 above) is not in the list, we will pick the new default value.

I think that this is fine for now? It doesn't seem that different from how other browsers work, as far as I can tell.

Reply:
I think the behaviour you describe is OK